### PR TITLE
Add experimental support for dynamic types, oracle consumer example

### PR DESCRIPTION
**Pull request by Netherdrake**

### CODE REVIEW

File: oracle_consumer/lib.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        /// This method is called from the Oracle's `callback` fn.
+        /// It can be named anything (in this case `set_bitcoin_price`),
+        /// however it does need a fixed selector.
+        /// The selector value needs to be the same as in the Oracle contract.
+        #[ink(message, selector = "0xB16B00B5")]
+        pub fn set_bitcoin_price(&mut self, result: OracleResult) -> Result<(),Error> {
+            // check if the oracle is trusted
+            let oracle = self.env().caller();
+            if oracle != self.authorized_oracle {
```
Comment (jbargu) [inline oracle_consumer/lib.rs:49]: This should include the owner of  the contract.

**Review by jbargu — COMMENTED** [review]
Although fine, I think the `OracleConsumer` contract should also have `init_request` method. However, I am not sure how doable it is right now and if it is not, ignore this comment.

I am mentioning this, since then we have the complete workflow: `init_request(oracle, callback_selector_method)` -> `Request` event emitted -> node executes the PQL -> the node calls the oracle `callback(result)` method.

I too wonder if we could leverage the off-chain worker capabilities to skip the trusted oracle contract altogether. In that case a worker would request the PQL data directly from the node and sign/commit it directly to the consumer contract.

**Comment by Netherdrake** [comment]
I did not add request support to keep the contract simple. Also the request does not need to originate in the same contract as callback.

As far as ocw goes, afaik it it is not possible to make it work with smart contracts. Also most chains don't have it. And lastly, even if it did work, this would limit the scalability as only blockchain validators could be the oracles then.

**Comment by Netherdrake** [comment]
Added OracleRequesterConsumer which can also do `init_request(...)`

**Pull request merged**